### ISSUE TITLE
How to enable train_mode

### ISSUE THREAD

**Issue by Jiawei-Yang** [MIC-DKFZ/nnUNet#170]
Dear Fabian,

Hi, I really appreciate your work! It is profoundly great!

Now, I'm exploring the dropout functionality in nnUNet.  I note that the function `predict_3D` in 

> nnunet/network_architecture/neural_network/SegmentationNetwork.py

mentioned `use_train_mode` is kinda broken. Have you fixed it already? I tried to force the model to be set in train model but uncomment that RuntimeError. The results turned out to be the same each time. How could I get different predictions with dropout turned on?

Thank you in advance for your kind help.

**Comment by Jiawei-Yang** [comment]
BTW, I modified the `predict_3D` function as follows:
        ```
if use_train_mode is not None and use_train_mode:
            for m in self.modules():
                if m.__class__.__name__.startswith('Dropout'):
                    print(m)
                    m.train()
                else:
                    m.eval()
```
However, that nnUnet still gave identical results should not be the case when dropout is enabled during inference, right?

**Comment by FabianIsensee** [comment]
you need to use Module.apply (https://pytorch.org/docs/stable/nn.html#torch.nn.Module.apply) for that
Best,
Fabian

**Comment by Jiawei-Yang** [comment]
> you need to use Module.apply (https://pytorch.org/docs/stable/nn.html#torch.nn.Module.apply) for that
> Best,
> Fabian

Yes, that solves my problem perfectly!

You're amazing, Fabian! The more I read about your code, the more I grow from it. 
Truly appreciate for your work.

Best,
Jiawei

**Comment by Jiawei-Yang** [comment]
Sorry, but I made a mistake before. What I compared were the results obtained from the best model obtained hours ago and the results obtained from the latest model. They were of course different. 

When I made the inference from the latest best mode, results remain the same as the inference runs 5 times. 

Here is my modification:
```
if use_train_mode is not None and use_train_mode:
           def set_train(m):
                if m.__class__.__name__.startswith('Dropout'):
                    print(m)
                    m.train()
                else:
                    m.eval()
            self.apply(set_train)
            print("I'm using train mode")
```

**Comment by FabianIsensee** [comment]
Hi,
I presume you have verified that the appropriate layers are in train mode? Have you tried to simply set the entire network to train? This should have the same effect (instance norm does the same thing for both train and test).
A part from that I don't know what else could be the problem. I have not tried to use dropout during inference for years now. 
Best,
Fabian

**Comment by wsskuleuven** [comment]
> Sorry, but I made a mistake before. What I compared were the results obtained from the best model obtained hours ago and the results obtained from the latest model. They were of course different.
> 
> When I made the inference from the latest best mode, results remain the same as the inference runs 5 times.
> 
> Here is my modification:
> 
> ```
> if use_train_mode is not None and use_train_mode:
>     def set_train(m):
>         if m.__class__.__name__.startswith('Dropout'):
>             print(m)
>             m.train()
>         else:
>             m.eval()
>     self.apply(set_train)
>         
> ```
> 
> It did print out the Dropout3D module

Hi Jiawei,

I was wondering have you successfully got different prediction results by using this modification? Thanks in advance.
